### PR TITLE
feat: add service level JWT support

### PR DESCRIPTION
**Pull request by octonato**
Still draft because depends on proxy release

Refs
- https://github.com/lightbend/kalix-proxy/pull/2107
- https://github.com/lightbend/kalix-proxy/releases/tag/v1.1.19

### CODE REVIEW

File: sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/ActionDescriptorFactory.scala [hunk 1]
```diff
@@ -189,26 +190,14 @@ private[impl] object ActionDescriptorFactory extends ComponentDescriptorFactory
...
-      val mergedOptions =
-        allOptions.flatten
-          .foldLeft(kalix.ServiceOptions.newBuilder()) { case (builder, serviceOptions) =>
-            builder.mergeFrom(serviceOptions)
-          }
-          .build()
-
-      // if builder produces the default one, we can returns a None
-      if (mergedOptions == kalix.ServiceOptions.getDefaultInstance) None
-      else Some(mergedOptions)
```
Comment (octonato) [inline sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/ActionDescriptorFactory.scala:210]: We had this code copied in two places and I needed a third one, so moved it into `ComponentDescriptorFactory.mergeServiceOptions`

File: sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/ComponentDescriptorFactory.scala [hunk 2]
```diff
@@ -48,19 +41,13 @@ import kalix.javasdk.impl.reflection.CombinedSubscriptionServiceMethod
...
 import org.springframework.web.bind.annotation.RequestMapping
-
 private[impl] object ComponentDescriptorFactory {
 
-  implicit class MethodOps(javaMethod: Method) {
-    def isPublic = Modifier.isPublic(javaMethod.getModifiers)
-
-    def hasAnnotation[A <: Annotation](implicit ev: ClassTag[A]) =
-      javaMethod.getAnnotation(ev.runtimeClass.asInstanceOf[Class[Annotation]]) != null
-  }
```
Comment (octonato) [inline sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/ComponentDescriptorFactory.scala:62]: Took the opportunity to do some clean-up. This moved to `Reflect.Syntax` and I also added a few more extenions

File: sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/ComponentDescriptorFactory.scala [hunk 3]
```diff
@@ -80,14 +66,10 @@ private[impl] object ComponentDescriptorFactory {
...
-    Modifier.isPublic(clazz.getModifiers) &&
-    clazz.getAnnotation(classOf[Subscribe.EventSourcedEntity]) != null
+    clazz.isPublic && clazz.hasAnnotation[Subscribe.EventSourcedEntity]
 
   def streamSubscription(clazz: Class[_]): Option[Subscribe.Stream] =
-    if (Modifier.isPublic(clazz.getModifiers))
-      Option(clazz.getAnnotation(classOf[Subscribe.Stream]))
-    else
-      None
+    clazz.getAnnotationOption[Subscribe.Stream]
```
Comment (octonato) [inline sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/ComponentDescriptorFactory.scala:72]: See `Reflect.Syntax`

File: sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/ComponentDescriptorFactory.scala [hunk 4]
```diff
@@ -70,8 +57,7 @@ private[impl] object ComponentDescriptorFactory {
     javaMethod.isPublic && javaMethod.hasAnnotation[Acl]
 
   def hasValueEntitySubscription(clazz: Class[_]): Boolean =
-    Modifier.isPublic(clazz.getModifiers) &&
-    clazz.getAnnotation(classOf[Subscribe.ValueEntity]) != null
+    clazz.isPublic && clazz.hasAnnotation[Subscribe.ValueEntity]
```
Comment (octonato) [inline sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/ComponentDescriptorFactory.scala:60]: See `Reflect.Syntax`

File: sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/JwtDescriptorFactory.scala [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import java.lang.reflect.Method
+import java.lang.reflect.Modifier
+
+import kalix.JwtMethodOptions
+import kalix.JwtServiceOptions
+import kalix.MethodOptions
+import kalix.javasdk.annotations.JWT
+import Reflect.Syntax._
+
+object JwtDescriptorFactory {
```
Comment (octonato) [inline sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/JwtDescriptorFactory.scala:28]: Moved all the JWT annotation logic to its own class. To follow what we did for ACL.

**Review by aludwiko — APPROVED** [review]
LGTM

**Pull request merged**